### PR TITLE
File module state=empty support

### PR DESCRIPTION
**Pull request by hudecof**
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
**files/file.py**

##### ANSIBLE VERSION
ansible 2.2.0 (devel 9d5dff5596) last updated 2016/09/25 12:34:18 (GMT +200)
  lib/ansible/modules/core: (devel e9d9af5b19) last updated 2016/11/17 23:39:37 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD 8aa338ddfa) last updated 2016/09/25 12:36:11 (GMT +200)
  config file = 
  configured module search path = Default w/o overrides
```

```

##### SUMMARY
https://github.com/ansible/ansible-modules-core/issues/902
People really need this.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @hudecof for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by Nindaleth** [comment]
Thanks for making this!

File: files/file.py [hunk 1]
```diff
@@ -444,6 +447,47 @@ def main():
 
         module.exit_json(dest=path, changed=True, diff=diff)
 
+    elif state == 'empty':
+        changed = False
+        print("XXXXXX: %s" % prev_state)
```
Comment (Nindaleth) [inline files/file.py]: Is this intended to stay here? Where will the output go?

**Comment by hudecof** [comment]
@Nindaleth thanks, forgotten debug line

**Comment by bcoca** [comment]
Question About This One
=======================

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us that you've taken time to contribute.

Unfortunately, we're not sure if we want this feature in the program, and I don't want this to seem confrontational.It is not something we want to add it since workarounds exist and the file module does not seem the appropriate place for it.

However, we're absolutely always up for discussion.  Since this is a really busy project, we don't always see comments on closed tickets, but want to encourage
open dialog.  You can stop by the development list, and we'd be glad to talk about it - and we might even be persuaded otherwise!

   * https://groups.google.com/forum/#!forum/ansible-devel

In the future, sometimes starting a discussion on the development list prior to implenting a feature can make getting things included a little easier, but it's not always necessary.

Thank you once again for this and your interest in Ansible!

**Comment by hudecof** [comment]
@bcoca see the +1 comments. This is really required feature, the workaround here is really not in place. There's still could be written library and uploaded to the galaxy, but I think this is essential feature.

**Pull request closed without merging**